go/database/vt/state_test.go

```diff
@@ -0,0 +1,417 @@
+		return state.GetStorage(address, key)
+	})
+}
+
+func (s *comparingState) GetCode(address common.Address) ([]byte, error) {
+	s.t.Helper()
+	b, err := getCmpStateValue(s, func(name string, state state.State) (immutable.Bytes, error) {
+		b, err := state.GetCode(address)
+		return immutable.NewBytes(b), err
+	})
+	return b.ToBytes(), err
+}
+
+func (s *comparingState) GetCodeSize(address common.Address) (int, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (int, error) {
+		return state.GetCodeSize(address)
+	})
+}
+
+func (s *comparingState) GetCodeHash(address common.Address) (common.Hash, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (common.Hash, error) {
+		return state.GetCodeHash(address)
+	})
+}
+
+func (s *comparingState) GetHash() (common.Hash, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (common.Hash, error) {
+		return state.GetHash()
+	})
+}
+
+func (s *comparingState) GetBalance(address common.Address) (amount.Amount, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (amount.Amount, error) {
+		return state.GetBalance(address)
+	})
+}
+
+func (s *comparingState) Apply(block uint64, update common.Update) error {
+	s.t.Helper()
+	return s.action(func(name string, state state.State) error {
+		return state.Apply(block, update)
+	})
+}
+
+//
+//	I/O features
+//
+
+func (s *comparingState) Flush() error {
+	s.t.Helper()
+	return s.action(func(name string, state state.State) error {
+		return state.Flush()
+	})
+}
+
+func (s *comparingState) Close() error {
+	s.t.Helper()
+	return s.action(func(name string, state state.State) error {
+		return state.Close()
+	})
+}
+
+// action is a helper function that iterates over all states in the comparingState
+// and applies the provided function to each state. It collects errors
+// and returns a single error that combines all errors encountered during
+// the execution of the provided function on each state.
+func (s *comparingState) action(do func(name string, state state.State) error) error {
+	s.t.Helper()
+	var errs []error
+	for name, state := range s.states {
+		if err := do(name, state); err != nil {
+			errs = append(errs, fmt.Errorf("state %s: %w", name, err))
+		}
+	}
+
+	return errors.Join(errs...)
+}
+
+// getCmpStateValue is a helper function that iterates over all states in the comparingState
+// and applies the provided function to each state. It collects errors and checks
+// for value consistency across all states. If all states return the same value,
+// it returns that value; otherwise, it returns an error indicating the mismatch.
+func getCmpStateValue[T comparable](s *comparingState, do func(name string, state state.State) (T, error)) (T, error) {
+	s.t.Helper()
+	var errs []error
+	var ref *T
+	var refName string
+	for name, state := range s.states {
```

Comment:
Since iteration through maps is random, this will result in the reference version being a different version in every call of `getCmpStateValue`. This could be an annoying property when debugging failing tests.

Can you make a fixed implementation the reference implementation?